bitcoin/test/run-tx-encode.c

```diff
@@ -12,7 +12,7 @@ static void hexeq(const void *p, size_t len, const char *hex)
 {
 	char *tmphex = tal_hexstr(NULL, p, len);
 
-	if (strcmp(hex, tmphex)) {
+	if (strcmp(hex, tmphex) != 0) {
```

Comment:
`ccan/str/str.h` has a `streq`, might be better here (unless this test does not want to pull in ccan?)